xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp

```diff
@@ -0,0 +1,1139 @@
+    }
+  }
+
+  /* disable add-ons that failed to initialise */
+  if (disableAddons.size() > 0)
+  {
+    CSingleLock lock(m_critUpdateSection);
+    for (VECADDONS::iterator it = disableAddons.begin(); it != disableAddons.end(); it++)
+    {
+      /* disable in the add-on db */
+      CAddonMgr::Get().DisableAddon((*it)->ID(), true);
+
+      /* remove from the audio dsp add-on list */
+      VECADDONS::iterator addonPtr = std::find(m_Addons.begin(), m_Addons.end(), *it);
+      if (addonPtr != m_Addons.end())
+        m_Addons.erase(addonPtr);
+    }
+  }
+
+  return bReturn;
+}
+
+bool CActiveAEDSP::UpdateAddons(void)
+{
+  VECADDONS addons;
+  bool bReturn(CAddonMgr::Get().GetAddons(ADDON_ADSPDLL, addons, true));
+  size_t usableAddons;
+
+  if (bReturn)
+  {
+    CSingleLock lock(m_critUpdateSection);
+    m_Addons = addons;
+  }
+
+  usableAddons = m_Addons.size();
+
+  /* handle "new" addons which aren't yet in the db - these have to be added first */
+  for (unsigned iAddonPtr = 0; iAddonPtr < m_Addons.size(); iAddonPtr++)
+  {
+    const AddonPtr dspAddon = m_Addons.at(iAddonPtr);
+
+    bool newRegistration = false;
+    if (RegisterAudioDSPAddon(dspAddon, &newRegistration) < 0 || newRegistration)
+    {
+      CAddonMgr::Get().DisableAddon(dspAddon->ID(), true);
+      usableAddons--;
+    }
+  }
+
+  if ((!bReturn || usableAddons == 0) && !m_bNoAddonWarningDisplayed &&
+      !CAddonMgr::Get().HasAddons(ADDON_ADSPDLL, false) &&
+      IsActivated())
+  {
+    m_bNoAddonWarningDisplayed = true;
+    CSettings::Get().SetBool("audiooutput.dspaddonsenabled", false);
+    CGUIDialogOK::ShowAndGetInput(24055, 24056, 24057, 24058);
+    CGUIMessage msg(GUI_MSG_UPDATE, WINDOW_SETTINGS_SYSTEM, 0);
+    g_windowManager.SendThreadMessage(msg, WINDOW_SETTINGS_SYSTEM);
+  }
+
+  return bReturn;
+}
+
+void CActiveAEDSP::Notify(const Observable &obs, const ObservableMessage msg)
+{
+  if (msg == ObservableMessageAddons)
+    UpdateAddons();
+}
+
+void CActiveAEDSP::Process(void)
+{
+  bool bCheckedEnabledAddonsOnStartup(false);
+
+  CAddonMgr::Get().RegisterAddonMgrCallback(ADDON_ADSPDLL, this);
+  CAddonMgr::Get().RegisterObserver(this);
+
+  UpdateAddons();
+
+  m_isActive = true;
+
+  while (!g_application.m_bStop && !m_bStop)
+  {
+    UpdateAndInitialiseAudioDSPAddons();
+
+    if (!bCheckedEnabledAddonsOnStartup)
+    {
+      bCheckedEnabledAddonsOnStartup = true;
+      if (HasEnabledAudioDSPAddons())
+        TriggerModeUpdate(false);
+      else if (!m_bNoAddonWarningDisplayed)
+        ShowDialogNoAddonsEnabled();
+    }
+
+    Sleep(1000);
+  }
+
+  m_isActive = false;
+}
+
+void CActiveAEDSP::ShowDialogNoAddonsEnabled(void)
+{
+  if (!IsActivated())
+    return;
+
+  CGUIDialogOK::ShowAndGetInput(15048, 15049, 0, 0);
+
+  vector<string> params;
+  params.push_back("addons://disabled/xbmc.adspaddon");
+  params.push_back("return");
+  g_windowManager.ActivateWindow(WINDOW_ADDON_BROWSER, params);
+}
```

Comment:
Don't hardcode that url in here. If I understood this method correctly, you want something like the "get more" functionality used for plugins. Extend that.